scanner/src/main/kotlin/scanners/fossid/FossIdNamingProvider.kt

```diff
@@ -47,26 +49,33 @@ class FossIdNamingProvider(
     }
 
     fun createProjectCode(projectName: String): String = namingProjectPattern?.let {
-        replaceNamingConventionVariables(namingProjectPattern, projectName, namingConventionVariables)
+        val builtins = mapOf(
+            "#projectName" to projectName
+        )
+        replaceNamingConventionVariables(namingProjectPattern, builtins, namingConventionVariables)
     } ?: projectName
 
-    fun createScanCode(projectName: String): String {
-        val pattern = namingScanPattern ?: "${projectName}_#currentTimestamp"
-        return replaceNamingConventionVariables(pattern, projectName, namingConventionVariables)
+    fun createScanCode(projectName: String, deltaTag: FossId.DeltaTag? = null): String {
+        val defaultPattern = "#projectName_#currentTimestamp${if (deltaTag == null) "" else "_#deltaTag"}"
```

Comment:
Thanks for the pattern fix, but ideally it would go to a separate commit (the first) in this PR, as it has been wrong before, and it's an independent change.